pkg/chunked/filesystem_linux_test.go

```diff
@@ -0,0 +1,339 @@
+package chunked
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/containers/storage/pkg/archive"
+	"github.com/containers/storage/pkg/chunked/internal"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type nopCloser struct {
+	*bytes.Reader
+}
+
+func (nopCloser) Close() error {
+	return nil
+}
+
+func TestSeekableFileGetBlobAt(t *testing.T) {
+	content := []byte("Hello, World!")
+
+	br := bytes.NewReader(content)
+	reader := nopCloser{br}
+
+	sf := newSeekableFile(reader)
+	chunks := []ImageSourceChunk{
+		{Offset: 0, Length: 5},
+		{Offset: 7, Length: 5},
+	}
+
+	streams, errs, err := sf.GetBlobAt(chunks)
+	assert.NoError(t, err)
+
+	expectedContents := [][]byte{
+		[]byte("Hello"),
+		[]byte("World"),
+	}
+
+	i := 0
+	for stream := range streams {
+		buf := new(bytes.Buffer)
+		_, err := buf.ReadFrom(stream)
+		require.NoError(t, err)
+
+		require.Equal(t, buf.Bytes(), expectedContents[i])
```

Comment:
(FWIW, order is `Equal(t, expected, actual)`. Not generally worth worrying about, that that can be cleaned up if the test ever fails.)

Reply:
fixed now